libs/domain/search/box/box.component.ts

```diff
@@ -0,0 +1,382 @@
+import { query } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { when } from 'lit/directives/when.js';
+import { html } from 'lit/static-html.js';
+import { SearchBoxOptions, SearchBoxProperties } from './box.model';
+import { baseStyles, searchboxStyles } from './styles';
+
+interface LinksSection {
+  title?: DirectiveResult;
+  options: SuggestionResource[];
+  type: SemanticLinkType;
+}
+
+@defaultOptions({
+  minChars: 2,
+  completionsCount: 5,
+  productsCount: 5,
+  categoriesCount: 5,
+  cmsCount: 0,
+})
+@hydratable(['mouseover', 'focusin'])
+@signalAware()
+export class SearchBoxComponent
+  extends ContentMixin<SearchBoxOptions>(LitElement)
+  implements SearchBoxProperties
+{
+  static styles = [baseStyles, searchboxStyles];
+
+  constructor() {
+    super();
+    this.onScroll = debounce(this.onScroll.bind(this), 20);
+    this.onTypeahead = debounce(this.onTypeahead.bind(this), 300);
+    this.onClear = this.onClear.bind(this);
+    this.onClose = this.onClose.bind(this);
+  }
+
+  @signalProperty() query = '';
+
+  @query('oryx-typeahead') typeahead!: TypeaheadComponent;
+  @query('div[slot="option"] > div') scrollContainer?: HTMLElement;
+
+  protected suggestionService = resolve(SuggestionService);
+  protected routerService = resolve(RouterService);
+  protected semanticLinkService = resolve(SemanticLinkService);
+  protected i18nService = resolve(I18nService);
+
+  protected $suggestion = computed(() => {
+    const options = this.$options();
+    const withSuggestion =
+      this.query &&
+      (!options.minChars || this.query.length >= options.minChars);
+    const getSuggestions = () => {
+      const raw = signal(this.suggestionService.get({ query: this.query }))();
+
+      return raw
+        ? {
+            completion: raw.completion.slice(0, options.completionsCount),
+            products: raw.products?.slice(0, options.productsCount) ?? [],
+            categories: raw.categories.slice(0, options.categoriesCount),
+            cmsPages: raw.cmsPages.slice(0, options.cmsCount),
+          }
+        : null;
+    };
+    const suggestion = withSuggestion ? getSuggestions() : null;
+
+    this.toggleAttribute?.('stretched', this.hasCompleteData(suggestion));
+
+    return suggestion;
+  });
+
+  protected $placeholder = signal(
+    this.i18nService.translate(['search', 'search.placeholder'])
+  );
+
+  protected $link = computed(() =>
+    this.semanticLinkService.get({
+      type: SemanticLinkType.ProductList,
+      ...(this.query ? { params: { q: this.query } } : {}),
+    })
+  );
+
+  connectedCallback(): void {
+    super.connectedCallback();
+
+    this.addEventListener('oryx.clear', this.onClear);
+    this.addEventListener('oryx.close', this.onClose);
+  }
+
+  disconnectedCallback(): void {
+    this.removeEventListener('oryx.clear', this.onClear);
+    this.removeEventListener('oryx.close', this.onClose);
+
+    super.disconnectedCallback();
+  }
```

Comment:
seems not needed

Reply:
I was thinking that we intentionally add `oryx.clear, oryx.close` api. Ok, I'll drop it